### PR TITLE
feat: student deletion

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Delete Student](https://www.notion.so/uwblueprintexecs/Delete-Student-92b89bb738f348b7ac64f3272f26079a)

## Implementation description
- Add mutation for deleting students
- Add student deletion action to popover button
- Disable deletion/editing of students from inactive classes

## Steps to test
1. Visit classroom drilldown page and test student deletion.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/teacher/student-management/view-students/StudentsTable.tsx [hunk 1]
```diff
@@ -8,15 +8,21 @@ import EditStudentPopover from "./EditStudentPopover";
...
   students,
+  classId,
+  isClassActive,
 }: StudentsTableProps): React.ReactElement => {
   const headers = ["First Name", "Last Name", "Student ID"];
   const rows: TableRow[] = students.map((student) => ({
     values: [student.firstName, student.lastName, student.studentNumber],
-    menu: <EditStudentPopover />,
+    menu: isClassActive ? (
+      <EditStudentPopover classId={classId} student={student} />
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/view-students/StudentsTable.tsx]: The name `EditStudentPopover` sounds like its only for edit, but idk what else you could call it. ManageStudentPopover sounds weird

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx [hunk 2]
```diff
@@ -1,18 +1,54 @@
...
   return (
-    <Popover isOpen={isOpen} onClose={onClose} onOpen={onOpen}>
-      <VStack divider={<Divider />} spacing="0em">
-        <PopoverButton name="Edit" onClick={() => {}} />
-        <PopoverButton name="Delete" onClick={() => {}} />
-      </VStack>
-    </Popover>
+    <>
+      <Popover isOpen={isOpen} onClose={onClose} onOpen={onOpen}>
+        <VStack divider={<Divider />} spacing="0em">
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]:
```suggestion
        <VStack divider={<Divider />} spacing={0}>
```

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx [hunk 3]
```diff
@@ -1,18 +1,54 @@
...
+          <PopoverButton name="Edit" onClick={() => {}} />
+          <PopoverButton name="Delete" onClick={openDeleteModal} />
+        </VStack>
+      </Popover>
+
+      <Modal
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: modal should be red?

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx [hunk 4]
```diff
@@ -1,18 +1,54 @@
...
+type EditStudentPopoverProps = {
+  student: StudentResponse;
+  classId: string;
+};
+
+const EditStudentPopover = ({
+  student,
+  classId,
+}: EditStudentPopoverProps): React.ReactElement => {
+  const [deleteStudent] = useMutation(DELETE_STUDENT, {
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: can we add success / error toast?

**Review by carissa-tang — COMMENTED** [review]
sidenote:

can we add padding to the bottom? also the space above search bar looks too big but double check our other search bars
image

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx (hunk 4 above)
Comment (jfdoming) [reply frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: I think I'll make this change in a follow-up pr. I found a few others that are missing toasts, and a few more that don't use standard modal components. Planning to refactor everything to use the same component and make an easier way of having these toasts

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx (hunk 3 above)
Comment (jfdoming) [reply frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: Yes oops noticed this too late

**Comment by jfdoming** [comment]
> sidenote:
> 
> can we add padding to the bottom? also the space above search bar looks too big but double check our other search bars image

To the bottom of what sorry?

**Comment by carissa-tang** [comment]
> > sidenote:
> > can we add padding to the bottom? also the space above search bar looks too big but double check our other search bars image
> 
> To the bottom of what sorry?

bottom of the empty state thing - my screenshot goes until the end of my screen

File: frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx (hunk 4 above)
Comment (carissa-tang) [reply frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: sure - can you make a ticket for it?

Comment (jfdoming) [reply frontend/src/components/teacher/student-management/view-students/EditStudentPopover.tsx]: https://www.notion.so/uwblueprintexecs/Add-loading-state-toasts-to-all-modals-7d8198e55ee545ebbe1f096740937b67

File: frontend/src/components/teacher/student-management/classroom-summary/DeleteClassroomModal.tsx [hunk 5]
```diff
@@ -13,19 +13,14 @@ const DeleteClassroomModal = ({
   onClose,
   deleteClassroom,
 }: DeleteModalProps): React.ReactElement => {
-  const onDeleteClassroom = async () => {
-    await deleteClassroom();
-    onClose();
```
Comment (carissa-tang) [inline frontend/src/components/teacher/student-management/classroom-summary/DeleteClassroomModal.tsx:18]: I think we still need this `onClose()` - right now the modal doesn't close after deletion

**Review by carissa-tang — APPROVED** [review]
approved but one quick thing

File: frontend/src/components/teacher/student-management/classroom-summary/DeleteClassroomModal.tsx (hunk 5 above)
Comment (jfdoming) [reply frontend/src/components/teacher/student-management/classroom-summary/DeleteClassroomModal.tsx:18]: hmm, it closes for me @carissa-tang ?

Comment (jfdoming) [reply frontend/src/components/teacher/student-management/classroom-summary/DeleteClassroomModal.tsx:18]: Aug-28-2023 00-17-57

**Review by carissa-tang — APPROVED** [review]
yay !

**Pull request merged**